source/OrderedTypes/PredicativeLFP.lagda

```diff
@@ -0,0 +1,1711 @@
+open PropositionalTruncation pt
+
+\end{code}
+
+We commence by defining what it means for a monotone endomap to have
+a least fixed point.
+
+\begin{code}
+
+module _ {𝓤 𝓦 𝓥 : Universe} (L : Sup-Lattice 𝓤 𝓦 𝓥) where
+
+ has-least-fixed-point : (f : ⟨ L ⟩ → ⟨ L ⟩) → 𝓤 ⊔ 𝓦  ̇
+ has-least-fixed-point f =
+  Σ p ꞉ ⟨ L ⟩ , (f p ＝ p) × ((a : ⟨ L ⟩) → (f a ＝ a) → (p ≤⟨ L ⟩ a) holds)
+
+ has-least-fixed-point-is-prop : (f : ⟨ L ⟩ → ⟨ L ⟩)
+                               → is-prop (has-least-fixed-point f) 
+ has-least-fixed-point-is-prop f (p₁ , fp₁ , l₁) (p₂ , fp₂ , l₂) =
+  to-subtype-＝ (λ x → ×-is-prop
+                       (sethood-of L)
+                       (Π-is-prop fe (λ y → Π-is-prop fe
+                        (λ _ → holds-is-prop (x ≤⟨ L ⟩ y)))))
+                (antisymmetry-of L (l₁ p₂ fp₂) (l₂ p₁ fp₁))
+
+\end{code}
+
+We construct the least closed subset of an inductive definition as a QIT family.
+Since QITs (and more generally HITs) are not native in Agda we will instead
+assume the existence of such a type as well as its induction principle.
+Technically speaking we are going to use record types to package the contents
+of this QIT family.
+
+Notice all constructors are 'strictly positive' and the path constructors are
+'natural' (see Chapter 6 Section 13 of the HoTT book
+https://homotopytypetheory.org/book/).
+
+Notice that the QIT family has two constructors which represent the closure
+conditions we wish to impose on subsets. The c-closure condition says:
+for any subset contained in the least closed subset, elements in the downset of
+its join are in the least closed subset as well. That is, Y is c-closed if
+  for any U ⊆ Y we have ↓ᴮ (⋁ U) ⊆ Y.
+The ϕ-cl constructor says:
+if for any a : L and b : B with (b , a) ∈ ϕ and ↓ᴮ a 'contained' in the least
+closed subset then b is in the least closed subset. That is, Y is ϕ-closed if
+  for any a : L and b : B we have ↓ᴮ a ⊆ Y ⇒ b ∈ Y.
+
+It is worth noting that we don't encode the downsets as subsets in type
+theory (rather they are total spaces) so for that reason we won't encode the
+closure conditions exactly as above (maybe add some notation to allow for
+more familiar form).
+
+We also derive the initiality of the least closed subset from the postulated
+induction principle. Initiality is closely related to the 'least' part of
+our least fixed point theorem.
+
+\begin{code}
+
+module _
+        {𝓤 𝓦 𝓥 : Universe}
+        {B : 𝓥  ̇}
+        (L : Sup-Lattice 𝓤 𝓦 𝓥)
+        (β : B → ⟨ L ⟩)
+        (h : is-small-basis L β)
+       where
+
+ private
+  _≤_ = order-of L
+  ⋁_ = join-of L
+
+ open Joins _≤_
+ open is-small-basis h
+
+\end{code}
+
+We start by giving names to the (dependent) closure conditions.
+
+\begin{code}
+
+ c-closure : {𝓣 : Universe} (S : 𝓟 {𝓣} B) → (𝓥 ⁺) ⊔ 𝓣  ̇
+ c-closure S = (U : 𝓟 {𝓥} B)
+             → U ⊆ S
+             → (b : B) → b ≤ᴮ (⋁ 【 β , U 】)
+             → b ∈ S
+
+ dep-c-closure : {𝓣 𝓣' : Universe} (S : 𝓟 {𝓣} B)
```

Comment:
I don't quite understand these dependent closure conditions to be honest. For one thing, it's a bit strange to see the powerset of (b ∈ S) which is the powerset of a proposition?

Would it not be more transparent to just have the record `inductively-generated-subset-exists` say that `𝓘` is the least closed subset, i.e. `𝓘` is initial? This condition is very clear and easy to understand. Do you need the induction principle/closure conditions for anything else other deriving this initiality?

Apologies that I didn't spot this sooner by the way!

Reply:
@tomdjong The dependent closure conditions are essentially just a copy of the individual clauses recorded from the induction principle. It does seem a bit strange to have the powerset of a propsition; it simply results from naively creating an induction principle for this degenerate type. In some sense the induction principle is overkill. 

No worries! I had a similar thought on my previous iteration of edits, but felt uneasy about scrapping the entire induction principle. Honestly, now it does seem to be the best approach.